Fix GLAD-related build problems on Linux

queezle42 opened the pull request:
See discussion in https://github.com/godotengine/godot/issues/69033

This PR replaces the remaining includes of `<GL/gl.h>` with `"thirdparty/glad/glad/gl.h"` and adds missing extensions (`EXT_texture` and `EXT_texture_sRGB`, required by the OpenXR-module) to the GLAD loader.

Tested on Linux (NixOS); should only affect Linux builds.

*Bugsquad edit:*
- Fixes #69033

queezle42 committed:
Fix GLAD-related build problems on Linux

- Use gl.h provided by GLAD in the OpenXR module
- Use non-EXT variants of some OpenGL defines
- Remove libGL-related code paths

akien-mga commented:
Looks good, but many of these `_EXT` defines are just the same as their non-`EXT` counterpart. I wonder if the problem isn't that we're using `_EXT` defines in places where what we need is already core?

queezle42 commented:
@akien-mga That works for everything, except `GL_RGB2_EXT` (from `EXT_texture`).

queezle42 commented:
I've removed `EXT_texture_sRGB` from GLAD and updated the commit to use non-`_EXT` defines where possible. `EXT_texture` is still required, unless there is a way to get rid of `GL_RGB2_EXT`.

clayjohn commented:
> I've removed `EXT_texture_sRGB` from GLAD and updated the PR to use non-`_EXT` defines where possible. `EXT_texture` is still required, unless there is a way to get rid of `GL_RGB2_EXT`.

I think we can just get rid of ``GL_RGB2_EXT`` as it isn't exposed for Android or Windows. It was likely only included because it was available (cc @dsnopek) . I would combine the #ifdef WINDOWS section with the #else section and then get rid of the extensions in GLAD.

queezle42 commented:
@clayjohn I think that is correct.

Since that function (`get_swapchain_format_name`) only maps the format name but the format is not used anywhere else in the engine it can probably be removed without causing problems. I have updated the PR again.

akien-mga commented on modules/openxr/extensions/openxr_opengl_extension.cpp:
```diff
@@ -167,8 +167,8 @@ void OpenXROpenGLExtension::get_usable_swapchain_formats(Vector<int64_t> &p_usab
 #else
-	p_usable_swap_chains.push_back(GL_SRGB8_ALPHA8_EXT);
-	p_usable_swap_chains.push_back(GL_RGBA8_EXT);
+	p_usable_swap_chains.push_back(GL_SRGB8_ALPHA8);
+	p_usable_swap_chains.push_back(GL_RGBA8);
```
The three `#if`/`#elif`/`#else` seem pretty redundant now since it's all the same code.

queezle42 commented:
I removed the libGL-related code path as suggested by @clayjohn and cleaned up the redundant ifdef.

dsnopek commented:
> I think we can just get rid of GL_RGB2_EXT as it isn't exposed for Android or Windows. It was likely only included because it was available (cc @dsnopek)

Yeah, we don't actually need it, we just put all the available formats on that list so we can print out the list of formats that an OpenXR runtime reports as supported. It should be totally fine to drop it!

dsnopek reviewed (approved):
The code looks good to me! Assuming this compiles on both Linux and Windows (CI is running now), I think it should be good.

Thanks! :-)

dsnopek commented:
Oh, but I just noticed that this has multiple commits. It'll need to get squashed down to a single commit before it can be merged:

https://docs.godotengine.org/en/stable/community/contributing/pr_workflow.html#the-interactive-rebase

queezle42 commented:
@dsnopek Done

akien-mga commented:
Thanks!
